### PR TITLE
Задача 6. Компьютерное зрение: задача сегментации

### PR DESCRIPTION
**Pull request by IAmFunkyFrog**
- [X] Сделать train/test split (или кросс-валидацию), зафиксировать все сиды чтобы более-менее воспроизводились результаты. Добавить метрику **jaccard index**. Переобучиться на одном батче (например 4 картинки), попробовать получить почти идеальное качество, посмотреть получилось или нет. Посчитать метрики для батча и для теста. Обучить на всем train, построить train/val кривые для loss и для **jaccard index**, подобрать оптимальное время обучения (количество шагов) для фиксированных параметров. Визуализировать topN самых плохих предсказаний.
- [X] Попробовать улучшить качество не меняя архитектуру: аугментации, гиперпараметры.
- [X] Попробовать улучшить качество изменяя архитектуру.

### CODE REVIEW

**Comment by VladimirGl** [comment]
посольку все выглядит так же как и у https://github.com/artmotika/ml-course/pull/2 к вам те же комментарии:

сложно графики читать, loss и jaccard в разных масштабах существуют, сравнивать нам их между собой смысла нету - так что лучше отдельно график где train и val jaccard и график где train и val loss

добавили ауги - трансформ на картинку и маску применяется по отдельности - значит картинка может повернуться а маска нет => даем не правильный семпл в обучение (маска не соответствует картинке) => эт приводит к тому что лосс хуже чем без аугов :)
поворот на до 180 (да и до 90) не оч имеет смысл для человеческих фоток ; небольшой да, а вверх ногами мы может и не ожидаем картинки (а может и ожидаем)
ауги которые не релевантны тесту - могут ухудшать качество
но любопытно что скор все равно адекватный
всегда имеет смысл глазами смотреть на то что dataloader вам отдает - эт оч частая ошибка чуть не верно обработать данные и искать че не так с архитектурой / пайплайоном :)
